apps/prairielearn/src/lib/preact-html.ts

```diff
@@ -0,0 +1,12 @@
+import { render } from 'preact-render-to-string/jsx';
+
+import { type HtmlSafeString, unsafeHtml } from '@prairielearn/html';
+import type { VNode } from '@prairielearn/preact-cjs';
+
+// This function must live outside of `preact.tsx` so that it can be imported
+// on the client. `preact.tsx` imports `@prairielearn/compiled-assets`, which
+// cannot be bundled for the browser.
+
+export function renderHtml(vnode: VNode): HtmlSafeString {
+  return unsafeHtml(render(vnode, {}, { pretty: true, jsx: false }));
```

Comment:
I added `pretty: true` to make the output easier to read manually (useful for logging HTML responses during tests for debugging). I'm not sure if this has a performance impact, and Preact themselves don't say much about that. Might be worth some small benchmarks? Or maybe we only want to do this on the server in production?

Reply:
I'll just change this to production-only.